packages/hdwallet-native/src/binance.ts

```diff
@@ -88,7 +89,8 @@ export function MixinNativeBinanceWallet<TBase extends core.Constructor<NativeHD
           throw Error("Invalid permissions to sign for address")
         }
         const addressTo = msg.tx.msgs[0].outputs[0].address;
-        const amount = (msg.tx.msgs[0].inputs[0].coins[0].amount).toString();
+        // The Binance SDK takes amounts as decimal strings.
+        const amount = Big(msg.tx.msgs[0].inputs[0].coins[0].amount).div(Big(10).pow(8)).toString();
```

Comment:
has the data structure already been validated at this point? i.e., if it's malformed and `inputs` is empty, this will hard crash.

Also, I would recommend doing an `isNaN()` check prior to `toString()` and raise an error if it's NaN.

Reply:
good point majorhayes. if it's a flaw, it's an existing one. when we're talking about wallet functionality, amounts, addresses etc it might be desirable to just hard crash if something is off vs alternatives like signing the wrong amount.

Does Big have a `Big.pow10(n)`?

Reply:
IMHO hard crash good, malformed data structures bad. none of the structures are really validated consistently anywhere, honestly; that seems to be more of a systemic problem. FWIW on the NaN front, `big.js` is a subset of `bignumber.js`, and doesn't do NaN or infinities at all.